app/models/queue_item.rb

```diff
@@ -10,8 +12,22 @@ def category_name
   end
 
   def rating
-    review = Review.where(user: user, video: video).first
     review.rating if review
   end
+
+  def rating=(new_rating)
+    rev = review || Review.create(user: user, video: video)
+    rev.rating = new_rating
+    rev.rating = nil if new_rating == ''
+
+    if rev.valid? || !rev.errors.keys.include?(:rating) || rev.rating.nil?
+      rev.update_attribute(:rating, new_rating)
+      return true
```

Comment:
While this `return` does show intent very well, keep in mind that it is not required.